### PR TITLE
feat: Color scheme dropdown

### PR DESCRIPTION
**Pull request by AugustinSorel**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

fixing this issue: https://github.com/t3-oss/create-t3-app/issues/847

This pr changed the old theme toggle to a drop down with 3 options (os default, light and dark)

## Screenshots

Screenshot from 2023-02-20 21-36-07

💯

### CODE REVIEW

**Review by juliusmarminge — APPROVED** [review]
1. Change `Os Default` to `System`
2. The system doesn't actually use the systems theme, you'll need to setup a media listener for that.

**Review by juliusmarminge — CHANGES_REQUESTED** [review]
(meant to press this xd)

**Comment by juliusmarminge** [comment]
https://user-images.githubusercontent.com/51714798/220202881-6567e425-6f8c-4977-808b-79c439295e8c.mp4

File: www/src/components/navigation/ColorSchemeSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+type ThemesNames = (typeof themesOptions)[number]["name"];
+
+const findThemeByName = (name: ThemesNames) => {
+  return themesOptions.find((theme) => theme.name === name);
+};
+
+const themesOptions = [
+  {
+    name: "os default",
```
Comment (bai) [inline www/src/components/navigation/ColorSchemeSelect.tsx:12]:
```suggestion
    name: "System",
```

To me, `System` sounds more in line with what others do, e.g.:

Screenshot 2023-02-21 at 05 12 04

File: www/src/components/navigation/ColorSchemeSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      .addEventListener("change", (e) => {
+        if (e.matches) {
+          findThemeByName("dark")?.clickHandler();
+          setSelectedTheme("dark");
+        } else {
+          setSelectedTheme("light");
+          findThemeByName("light")?.clickHandler();
+        }
+      });
+  }, []);
```
Comment (juliusmarminge) [inline www/src/components/navigation/ColorSchemeSelect.tsx:87]: this event listener should only run when the system option is selected, we should also have a cleanup function

**Review by juliusmarminge — COMMENTED** [review]
Still some unexpected behaviors:

1. The icon shouldn't change when i change my system preferences, it should be on the system icon as long as i haven't changed it.

2. the button's focus state is triggered when i change the system preferences even though i haven't touched the button

3. If i have changed it, the theme shouldn't change even if i change the system preferences.

demo showcasing these faults:

https://user-images.githubusercontent.com/51714798/220986678-34144c18-060b-4339-8fcd-cdf2ff0565cc.mp4

**Comment by juliusmarminge** [comment]
Closing as stale so that others don't feel blocked to pick it up

**Pull request closed without merging**